lib/3scale/backend/listener_metrics.rb

```diff
@@ -93,6 +132,24 @@ def code_group(resp_code)
             'unknown'.freeze
           end
         end
+
+        def req_type(path)
+          AUTH_AND_REPORT_REQUEST_TYPES[path] || internal_api_req_type(path)
+        end
+
+        def internal_api_req_type(path)
+          (_regex, type) = INTERNAL_API_PATHS.find { |(regex, _)| regex.match path }
```

Comment:
This is very unfortunate. With this we need to match against regexes a path which Sinatra already matched for us, so double work, but this time much more expensive than matchings strings.

We could use a [Custom Route Matcher](http://sinatrarb.com/intro.html) although that seems a bit too much for this case (and it's not like it's greatly documented), or in a more practical way we could use `request.env['prometheus.iapi.req_type'] = 'alerts'` in the relevant Internal API endpoints?

Reply:
Your solution would be more efficient, that's true. However, it implies adding a similar line to the one you pasted in all the files that define the endpoints. Also, this is part of an optional feature, and it only affects the internal API, not the main one. For those reasons, I'd rather merge it as it is, and only change this if it proves to be a problem when we run our benchmarks.

Reply:
:+1: